docs/src/howto-guides/child_device_config_managament.md

```diff
@@ -0,0 +1,203 @@
+the `url` received in the request with an HTTP PUT request.
+
+Once the upload is complete, send a "successful" MQTT status message as follows:
+
+**Topic:**
+
+`tedge/{child-d}/commands/res/config_snapshot`
+
+**Payload**:
+
+```json
+{
+    "status": "successful",
+    "type": "{config-type}",
+    "path": "/child/local/fs/path" 
+}
+```
+
+If there are any failures while reading or uploading the requested conifg file,
+a "failed" status update must be sent instead, to the same topic as follows:
+
+```json
+{
+    "status": "failed",
+    "type": "{config-type}",
+    "path": "/child/local/fs/path" 
+}
+```
+
+# Handle config update requests from thin-edge
+
+Handling config update requests from thin-edge is a 5-step process:
+
+1. Subsribe to, and receive conifg update requests via MQTT
+1. Send an “executing” operation status update to acknowledge the receipt of the request via MQTT
+1. Download the config file update from the URL received in the request via HTTP
+1. Apply the config file update on the child device
+1. Send a “successful” operation status update via MQTT
+
+The child device agent must subscribe to the `tedge/{child-d}/commands/req/config_update` MQTT topic
+to receive the config update requests from thin-edge.
+These requests arrive in the following JSON format:
+
+```json
+{
+    "type": "{config-type}",
+    "path": "/child/local/fs/path",
+    "url": "http://{tedge-ip}:8000/tedge/file-transfer/{child-d}/config_update/{config-type}"
+}
+```
+
+The child device agent can download the conifg file update from thin-edge using the `url` for the given `type`.
+
+On receipt of the request, the agent must send an "executing" MQTT status message as follows:
+
+**Topic:**
+
+`tedge/{child-d}/commands/res/config_update`
+
+**Payload**:
+
+```json
+{
+    "status": "executing",
+    "type": "{config-type}",
+    "path": "/child/local/fs/path" 
+}
+```
+
+After sending this status message, the agent must download the configuration file update
+from the `url` received in the request with an HTTP GET request.
+The agent can then apply the downloaded configuration file update on the device.
+
+Once the update is applied, send a "successful" MQTT status message as follows:
+
+**Topic:**
+
+`tedge/{child-d}/commands/res/config_update`
+
+**Payload**:
+
+```json
+{
+    "status": "successful",
+    "type": "{config-type}",
+    "path": "/child/local/fs/path" 
+}
+```
+
+If there are any failures while downloading and applying the update,
+a "failed" status update must be sent instead, to the same topic as follows:
+
+```json
+{
+    "status": "failed",
+    "type": "{config-type}",
+    "path": "/child/local/fs/path" 
+}
+```
+
+## Reference Implementation
+
+[Here](https://github.com/thin-edge/thin-edge.io_spikes/blob/main/file-transfer-service/child_device.py) is a reference child device agent implementaion written in Python to demonstrate the above mentioned protocol.
```

Comment:
This example must move to the [example repository](https://github.com/thin-edge/thin-edge.io_examples),
the spike repository being private.

Reply:
Done in this PR: https://github.com/thin-edge/thin-edge.io_examples/pull/36